fixtures/bucket_books_tree.py

```diff
@@ -39,13 +41,27 @@ def bucket_books_tree(s3_base_url, s3_queue_state_bucket_books):
         f"{s3_base_url}{s3_archive_folder}{i}@{j}.json" for i, j in zip(uuid_list, version_list)
     ]
 
-    book_tree = []
+    # list of approved book urls without http errors
+    approved_books_full_url_modified = []
+
+    for ublu in approved_books_full_url:
 
-    for books in range(len(approved_books_full_url)):
+        try:
+            urllib.request.urlopen(ublu)
+        except HTTPError as h_e:
+            # Return code 404, 501, ...
+            print("HTTPError (check concourse jobs): {}".format(h_e.code) + ", " + ublu)
+        else:
+            # Return code 200
+            approved_books_full_url_modified.append(ublu)
+
+    book_tree = []
 
-        s3_books = urllib.request.urlopen(approved_books_full_url[books]).read()
+    for books in range(len(approved_books_full_url_modified)):
 
+        s3_books = urllib.request.urlopen(approved_books_full_url_modified[books]).read()
```

Comment:
Minor: I believe you can insert the results into `book_tree` in the `try` block while iterating on `approved_books_full_url` instead of using `approved_books_full_url_modified` as an intermediate list you iterate on in a second pass.

Reply:
@rnathuji I will have a look at this later once I am done with tests. For now I pushed changes regarding your first comment.